modules/core/src/main/java/org/apache/ignite/internal/visor/consistency/VisorConsistencyRepairTask.java

```diff
@@ -117,15 +117,25 @@ protected VisorConsistencyRepairJob(VisorConsistencyRepairTaskArg arg, boolean d
             log.info("Consistency check started " +
                 "[grp=" + grpCtx.cacheOrGroupName() + ", part=" + p + ", strategy=" + strategy + "]");
 
-            VisorConsistencyStatusTask.MAP.put(arg, "0/" + part.fullSize());
+            StringBuilder sb = new StringBuilder();
+
+            sb.append("[node=").append(ignite.localNode());
+            sb.append(", cacheGroup=").append(grpCtx.cacheOrGroupName());
+            sb.append(", part=").append(p).append("]");
+
+            String statusKey = sb.toString();
+
+            if (VisorConsistencyStatusTask.MAP.putIfAbsent(statusKey, "0/" + part.fullSize()) != null)
+                throw new IllegalStateException("Consistency check already started " +
+                    "[grp=" + grpCtx.cacheOrGroupName() + ", part=" + p + "]");
```

Comment:
"]" can be replaced with ']'

Reply:
This can be extended later, so it's a string, not a char.
Let's keep it as is?